lib/interfaces/include/IOExpander.h

```diff
@@ -0,0 +1,30 @@
+#ifndef IOEXPANDER_H
+#define IOEXPANDER_H
+
+#include <etl/singleton.h>
+#include "MCP23017.h"
+
+class IOExpander
+{
+public:
+    IOExpander(uint8_t addr);
+
+private:
+    MCP23017 mcp;
+    uint16_t data;
+    
+public:
+    /* Get data from IOExpander.
+    * Note this only updates the data read. To update any values to be used, please use getBit()
+    */
+    void read();
+    /* Gets data from specified bit from specified port
```

Comment:
nit: The "read() only reads the data!" warning is more applicable on the getBit() function, since you might intuitively expect THIS one to also read from the source.